zio-http/shared/src/main/scala/zio/http/endpoint/Endpoint.scala

```diff
@@ -340,9 +340,10 @@ final case class Endpoint[PathInput, Input, Err, Output, Auth <: AuthType](
               case Some(HttpCodecError.CustomError("SchemaTransformationFailure", message))
                   if maybeUnauthedResponse.isDefined && message.endsWith(" auth required") =>
                 maybeUnauthedResponse.get
-              case Some(_) =>
+              case Some(HttpCodecError.MissingAuthorizationHeader) =>
+                Handler.succeed(Response.unauthorized)
```

Comment:
Let me know if this should be reported the same way is any other missing header.

Reply:
@987Nabil the `AuthType` has quite limited information, please see the implementation [draft](https://github.com/zio/zio-http/pull/3281/commits/113c4975eeaafebfa0c39e3f02f9ff90f6a45c00). Any piece of advice here?

Reply:
Most things are optional and can be omitted. Digest nonce is generated anyway.
Basic Digest and OAuth need a realm. We can add this to `AuthType` in the same fashion like `ScopedAuth`.
In case there is no realm, we could return no header or a default value. I tend to a default value. Like `"default"` or `"restricted"`